barretenberg/cpp/src/barretenberg/stdlib/primitives/bigfield/bigfield.hpp

```diff
@@ -587,9 +589,16 @@ template <typename Builder, typename T> class bigfield {
      *
      * @return true if the bigfield is constant, false otherwise.
      *
-     * TODO(https://github.com/AztecProtocol/aztec-packages/issues/14662): should we check if all limbs are constants?
+     * @details We use assertions to ensure that all limbs are consistent in their constant status.
      */
-    bool is_constant() const { return prime_basis_limb.witness_index == IS_CONSTANT; }
+    bool is_constant() const
+    {
+        ASSERT(binary_basis_limbs[0].element.is_constant() == binary_basis_limbs[1].element.is_constant() &&
```

Comment:
This is slightly inefficient, since you are repeating 3 calls twice

Reply:
Right, made this change.